### PR TITLE
Rails training Step 9: I18n

### PR DESCRIPTION
**Pull request by KumoRaku**
やること：

> 日本語部分を共通化
> Railsのi18nの仕組みを利用して、日本語リソースの共通化をしましょう
> ※i18n化を行うと後のステップでメッセージを出したりするのが楽になるなどのメリットがあります
> タイムゾーンの設定
> Railsのタイムゾーンを日本（東京）に設定しましょう
> 

JIRA：
https://jira.rakuten-it.com/jira/browse/RAKUTRA-4819

### CODE REVIEW

File: myapp/app/views/tasks/show.html.erb [hunk 1]
```diff
@@ -1,11 +1,11 @@
-<h1>Task Detail</h1>
+<h1><%= t '.title' %></h1>
 
-<h2>Title: <%= @task.title %></h2>
-<p>Description: <%= @task.description %></p>
+<h2><%= @task.title %></h2>
+<p><%= (Task.human_attribute_name :description) + ":" +  @task.description %></p>
```
Comment (takumism) [inline myapp/app/views/tasks/show.html.erb]: I think it is better to use "string interpolation" when concatenating characters in ruby 👍 
https://linuxhint.com/ruby-string-interpolation/

```suggestion
<p><%= "#{Task.human_attribute_name :description}:#{@task.description}" %></p>
```

File: myapp/config/locales/model.ja.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        task:
+          id: ID
+          # view側： User.human_attribute_name :name => "名前" /　t("activerecord.attributes.user.name")と同じ
+          title: タイトル
+          description: 詳細
+
+  # 全てのmodelで共通して使用するattributesを定義
+  attributes:
+    created_at: 作成日
+    updated_at: 更新日
```
Comment (takumism) [inline myapp/config/locales/model.ja.yml]: POSIX compliance requires a new line at the end of the file!

File: myapp/app/views/tasks/edit.html.erb [hunk 3]
```diff
@@ -1,3 +1,3 @@
-<h1>Edit Task</h1>
+<h1><%= t '.title' %></h1>
```
Comment (takumism) [inline myapp/app/views/tasks/edit.html.erb]: 【FYI】
In Rakuma, a member uses this helper method by writing `i18n.t`!
`i18n` can be omitted, but we want to indicate explicitly that we are using i18n.

**Review by takumism — COMMENTED** [review]
@KumoRaku 
I commented on several points.
I see that the following requirements have not been implemented, so could you confirm this and implement it?


> エラーページの設定
Railsが用意しているデフォルトのエラーページを自分が作った画面にしてみましょう
状況に応じて、適切にエラーページを設定しましょう
ステータスコードの404ページと500ページの2種類の設定は少なくとも必須とします

https://github.com/Fablic/training/blob/develop/steps_jp.md#%E3%82%B9%E3%83%86%E3%83%83%E3%83%979-%E3%82%A2%E3%83%97%E3%83%AA%E3%81%AE%E5%90%84%E7%A8%AE%E8%A8%AD%E5%AE%9A%E3%82%92%E8%A1%8C%E3%81%84%E3%81%BE%E3%81%97%E3%82%87%E3%81%86

File: myapp/app/views/tasks/edit.html.erb (hunk 3 above)
Comment (KumoRaku) [reply myapp/app/views/tasks/edit.html.erb]: Thank you for telling me. That's good.
I'll use `I18n.t`

File: myapp/config/locales/model.ja.yml (hunk 2 above)
Comment (KumoRaku) [reply myapp/config/locales/model.ja.yml]: Thank you, I add a new line.

File: myapp/app/views/tasks/show.html.erb (hunk 1 above)
Comment (KumoRaku) [reply myapp/app/views/tasks/show.html.erb]: Thank you for telling me.
I fixed it

**Comment by KumoRaku** [comment]
@craftscat 
I fixed my code as your comments.
Could you please check again?
And for these contents

> エラーページの設定
> Railsが用意しているデフォルトのエラーページを自分が作った画面にしてみましょう
> 状況に応じて、適切にエラーページを設定しましょう
> ステータスコードの404ページと500ページの2種類の設定は少なくとも必須とします

I'll implement this in my next PR. Is that OK?

**Comment by takumism** [comment]
@KumoRaku 
OK, I'll check it !

> I'll implement this in my next PR. Is that OK?

It's OK, no problem.

**Review by takumism — COMMENTED** [review]
This PR's change is OK, but I found several points in the existing code that could be improved, so I will comment on them!

**Comment by takumism** [comment]
It does not need one indentation on lines 46-48!

https://github.com/Fablic/training/blob/6ec0984a1a345dea86b4b554e38bd379b6c28df5/myapp/app/controllers/tasks_controller.rb#L45-L48

```ruby
  private

  def task_params
    params.require(:task).permit(:title, :description)
  end
```

**Comment by takumism** [comment]
Wouldn't it be better to have error handling?
And as for the other actions, wouldn't it be more user-friendly to have a flash message?

https://github.com/Fablic/training/blob/6ec0984a1a345dea86b4b554e38bd379b6c28df5/myapp/app/controllers/tasks_controller.rb#L38-L43

**Comment by takumism** [comment]
Is it ok to allow null for these columns?
In my opinion, at least for title, it is better to specify a NOT NULL constraint.

https://github.com/Fablic/training/blob/6ec0984a1a345dea86b4b554e38bd379b6c28df5/myapp/db/schema.rb#L16-L17

**Comment by takumism** [comment]
Wouldn't it be better to have validation on each element of the Task model instance?
For example, task names can be 4~25 characters long.

https://github.com/Fablic/training/blob/6ec0984a1a345dea86b4b554e38bd379b6c28df5/myapp/app/models/task.rb#L1-L2

**Comment by takumism** [comment]
I think it is better to use single quotations !

https://github.com/Fablic/training/blob/6ec0984a1a345dea86b4b554e38bd379b6c28df5/myapp/app/views/tasks/edit.html.erb#L3

https://github.com/Fablic/training/blob/6ec0984a1a345dea86b4b554e38bd379b6c28df5/myapp/app/views/tasks/new.html.erb#L3

```ruby
<%= render 'form', task: @task %>
```

**Comment by takumism** [comment]
It's hard to check the detailed Ruby coding conventions and modify the code, so you should install linter!

> Gemfile で [RuboCop](https://github.com/rubocop/rubocop)（Rubyの静的コード解析ツール） か [fablicop](https://github.com/Fablic/fablicop)をインストールしましょう。設定やコマンド実行方法は各READMEから確認しましょう。

https://github.com/Fablic/training/blob/KumoRaku-Step9-I18n/myapp/docs/steps_jp.md#%E3%82%B9%E3%83%86%E3%83%83%E3%83%977-%E3%82%BF%E3%82%B9%E3%82%AF%E3%82%92%E7%99%BB%E9%8C%B2%E6%9B%B4%E6%96%B0%E5%89%8A%E9%99%A4%E3%81%A7%E3%81%8D%E3%82%8B%E3%82%88%E3%81%86%E3%81%AB%E3%81%97%E3%82%88%E3%81%86

**Comment by takumism** [comment]
`capybara`, `selenium-webdriver`, and `rspec-rails` are not likely to be used in the development environment, so it would be better to put them in the test group from line 48 !

https://github.com/Fablic/training/blob/6ec0984a1a345dea86b4b554e38bd379b6c28df5/myapp/Gemfile#L34-L36

**Comment by takumism** [comment]
【Optional】

It would be easier for review and future development if we predefine `Allow NULL`, `Uniqueness`, `Indexes` and so on for each table.

https://github.com/Fablic/training/blob/KumoRaku-Step9-I18n/myapp/README.md

**Comment by KumoRaku** [comment]
> It does not need one indentation on lines 46-48!

I'll fix this.

**Comment by KumoRaku** [comment]
> Wouldn't it be better to have error handling?

I'll do this in the PR which I implement error pages of Step9

**Comment by KumoRaku** [comment]
> Is it ok to allow null for these columns?
> In my opinion, at least for title, it is better to specify a NOT NULL constraint.

and
> Wouldn't it be better to have validation on each element of the Task model instance?
> For example, task names can be 4~25 characters long.

and
>It would be easier for review and future development if we predefine Allow NULL, Uniqueness, Indexes and so on for each table.

I'll do these in Step11

**Comment by KumoRaku** [comment]
> I think it is better to use single quotations !

and
> It's hard to check the detailed Ruby coding conventions and modify the code, so you should install linter!

and
> capybara, selenium-webdriver, and rspec-rails are not likely to be used in the development environment, so it would be better to put them in the test group from line 48 !

I'll fix these

**Review by takumism — APPROVED** [review]
Thank you for modifications!
LGTM!
LGTM

**Pull request merged**